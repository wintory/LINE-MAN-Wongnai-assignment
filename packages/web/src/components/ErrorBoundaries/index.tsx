import { Button, Grid, styled, Typography } from '@mui/material';
import { FC } from 'react';

const PageWrapper = styled('div')(() => ({
  width: '100vw',
  height: '100vh',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
}));

const ErrorBox = styled('div')(() => ({
  display: 'block',
  textAlign: 'center',
  padding: '1.6rem',
}));

const ErrorBoundaries: FC = () => {
  const handleBackToHome = () => {
    window.location.href = '/';
  };

  const handleRefreshPage = () => {
    window.location.reload();
  };

  return (
    <PageWrapper>
      <ErrorBox>
        <Typography variant="h2" color="common.black">
          Something went wrong
        </Typography>
        <Typography variant="h4" pt={1} mb={2}>
          Please either refresh page or return to home to try again.
        </Typography>
        <Grid display="grid">
          <Button
            color="primary"
            variant="contained"
            onClick={handleRefreshPage}
          >
            <Typography variant="h5">Back to Home</Typography>
          </Button>
          <Button
            color="primary"
            variant="outlined"
            onClick={handleBackToHome}
            sx={{ mt: 2 }}
          >
            <Typography variant="h5">Refresh Page</Typography>
          </Button>
        </Grid>
      </ErrorBox>
    </PageWrapper>
  );
};

export default ErrorBoundaries;
