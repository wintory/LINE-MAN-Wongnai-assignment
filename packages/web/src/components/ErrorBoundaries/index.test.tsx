import { render } from '@testing-library/react';
import ErrorBoundaries from '.';

describe('ErrorBoundaries', () => {
  beforeAll(() => {
    global.window = Object.create(window);
    Object.defineProperty(window, 'location', {
      value: {
        href: 'http://test.com',
        reload: jest.fn(),
      },
      writable: true,
    });
  });

  it('should render if it exists', () => {
    const { getByText } = render(<ErrorBoundaries />);

    expect(getByText('Something went wrong')).toBeTruthy();
    expect(
      getByText('Please either refresh page or return to home to try again.')
    ).toBeTruthy();
    expect(getByText('Refresh Page')).toBeTruthy();
    expect(getByText('Back to Home')).toBeTruthy();
  });

  it('should match snapshot', () => {
    const { container } = render(<ErrorBoundaries />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should call back to home', () => {
    const { getByText } = render(<ErrorBoundaries />);

    const btn = getByText('Back to Home');

    btn.click();
    expect(window.location.href).toEqual('/');
  });

  it('should call reload page', () => {
    const { getByText } = render(<ErrorBoundaries />);

    const btn = getByText('Refresh Page');

    btn.click();
    expect(window.location.reload).toHaveBeenCalled();
  });
});
