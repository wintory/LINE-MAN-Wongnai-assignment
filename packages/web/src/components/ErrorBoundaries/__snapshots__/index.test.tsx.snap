// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundaries should match snapshot 1`] = `
<div
  class="css-1wb36w"
>
  <div
    class="css-airixb"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2 css-1obuk2h-MuiTypography-root"
    >
      Something went wrong
    </h2>
    <h6
      class="MuiTypography-root MuiTypography-subtitle1 css-1da9m30-MuiTypography-root"
    >
      Please either refresh page or return to home to try again.
    </h6>
    <div
      class="MuiGrid-root css-1xmc6ts-MuiGrid-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
        >
          Back to Home
        </h6>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-6q3cg8-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
        >
          Refresh Page
        </h6>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
